### ISSUE TITLE
Error when scanning for pictures (large library, existing cache)

### ISSUE THREAD

**Issue by salamanders** [arsenetar/dupeguru#415]
Application Identifier: com.hardcoded-software.dupeguru
Application Version: 4.0.3
Mac OS X Version: Version 10.12.3 (Build 16D32)

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 111, in __getitem__
KeyError: 'path:/Users/MYSELF/Pictures/Sony/2015-10-09/DSC00788.JPG'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
  File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 121, in purge_outdated
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 113, in __getitem__
KeyError: b'path:/Users/MYSELF/Pictures/Sony/2015-10-09/DSC00788.JPG'

**Comment by novelero** [comment]
Application Identifier: com.hardcoded-software.dupeguru
Application Version: 4.0.3
Mac OS X Version: Version 10.12.4 (Build 16E195)

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
  File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 129, in purge_outdated
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 47, in __delitem__
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 128, in __delitem__
KeyError: b'id:34932'

**Comment by what2be** [comment]
Application Identifier: com.hardcoded-software.dupeguru
Application Version: 4.0.3
Mac OS X Version: Version 10.12.6 (Build 16G29)

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 111, in __getitem__
KeyError: 'path:/Users/macbookpro/Pictures/Photos Library.photoslibrary/resources/proxies/derivatives/33/00/3394/UNADJUSTEDNONRAW_thumb_3394.jpg'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
  File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
  File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
  File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
  File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 121, in purge_outdated
  File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 113, in __getitem__
KeyError: b'path:/Users/macbookpro/Pictures/Photos Library.photoslibrary/resources/proxies/derivatives/33/00/3394/UNADJUSTEDNONRAW_thumb_3394.jpg'

**Comment by apj68** [comment]
I get the same.  I'm a first time dupeGuru user so I was having a little trouble figuring out what I wanted to do.  I ran several scans successfully using the "Standard" type with different references.  These were 32gb scans.  About the third time I figured I knew what I wanted to do so I ran it again, this time using a Picture mode scan with again, changed references.  I'm stuck now because I keep getting this error.  I restarted and tried clearing the Picture cache but I kept the same error:

     Application Identifier: com.hardcoded-software.dupeguru
    Application Version: 4.0.3
    Mac OS X Version: Version 10.12.6 (Build 16G1036)

    Traceback (most recent call last):
      File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 111, in __getitem__
    KeyError: 'path:/Volumes/HOME/apj/BeerSmith/help/_derived/undo_edit.htm_cmp_glacier110_vbtn.gif'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "build/dupeGuru.app/Contents/Resources/py/cocoa/inter.py", line 259, in pulse
      File "build/dupeGuru.app/Contents/Resources/py/hscommon/gui/progress_window.py", line 101, in pulse
      File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 323, in _job_error
      File "build/dupeGuru.app/Contents/Resources/py/hscommon/jobprogress/performer.py", line 43, in _async_run
      File "build/dupeGuru.app/Contents/Resources/py/core/app.py", line 780, in do
      File "build/dupeGuru.app/Contents/Resources/py/core/scanner.py", line 137, in get_dupe_groups
      File "build/dupeGuru.app/Contents/Resources/py/core/pe/scanner.py", line 31, in _getmatches
      File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 167, in getmatches
      File "build/dupeGuru.app/Contents/Resources/py/core/pe/matchblock.py", line 65, in prepare_pictures
      File "build/dupeGuru.app/Contents/Resources/py/core/pe/cache_shelve.py", line 121, in purge_outdated
      File "build/dupeGuru.app/Contents/Resources/py/shelve.py", line 113, in __getitem__
    KeyError: b'path:/Volumes/HOME/apj/BeerSmith/help/_derived/undo_edit.htm_cmp_glacier110_vbtn.gif'

**Comment by apj68** [comment]
Additional info -- I did this again but removed all folders and restarted.  Then I added back only one reference and one normal folder and ran it again but I got the same error as my comment above -- which referenced the KeyError _in the same file_.  But that file _wasn't even in the scan_.  So I went in yet again and set the debug mode, thinking that I would get more data to put in this ticket.  After I did that, it hasn't failed yet!  Go figure!
